src/pet.js

```diff
@@ -0,0 +1,79 @@
+function Pet(petName) {
+    const initialFitness = 10;
+    this.name = petName;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = initialFitness;
+    this.children = [];
+};
+
+Pet.prototype = {
+    get isAlive() {
+    const unfit = 0;
+    const starving = 10;
+    const tooOld = 30;
+    return this.age < tooOld && this.hunger < starving && this.fitness > unfit;
+}
+};
+
+Pet.prototype.growUp = function() {
+    if (!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+    }
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+    return 'Back in my day...vets are not what they used to be';
+};
+
+Pet.prototype.walk = function() {
+    const maxFitness = 10;
+    if (!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+    }
+    this.fitness += 4;
+    if (this.fitness > maxFitness) {
+        this.fitness = maxFitness;
+    } else this.fitness = this.fitness;
+    return 'About time we went outside';
```

Comment:
This could be refactored a bit. In essence you are adding a value to the existing fitness, without allowing it go over a set limit. You can do this with Math.min, which will return the lowest of the values passed to it:
```
this.fitness = Math.min(this.fitness + 4, max_fitness);

return 'About time we went outside';
```